roles/openshift_registry/tasks/main.yml

```diff
@@ -12,3 +10,56 @@
     --credentials={{ openshift_master_config_dir }}/openshift-registry.kubeconfig {{ _oreg_images }}
   register: _oreg_results
   changed_when: "'service exists' not in _oreg_results.stdout"
+
+- name: Create temp directory for volume definitions
+  command: mktemp -d /tmp/openshift-ansible-XXXXXXX
+  register: mktemp
+  changed_when: False
+  when: attach_registry_volume | bool
+
+- name: Land registry PersistentVolume definition
+  template:
+    dest: "{{ mktemp.stdout }}/registry-volume.json"
+    src: registry-volume.json.j2
+  when: attach_registry_volume | bool
+
+- name: Land registry PersistentVolumeClaim definition
+  template:
+    dest: "{{ mktemp.stdout }}/registry-claim.json"
+    src: registry-claim.json.j2
+  when: attach_registry_volume | bool
+
+- name: Create registry PersistentVolume
+  command: >
+    {{ openshift.common.client_binary }} create -f {{ mktemp.stdout }}/registry-volume.json
+  register: pv_create_output
+  failed_when: ('already exists' not in pv_create_output.stderr if pv_create_output.stderr else False) or ('created' not in pv_create_output.stdout if pv_create_output.stdout else False)
+  when: attach_registry_volume | bool
+
+- name: Create registry PersistentVolumeClaim
+  command: >
+    {{ openshift.common.client_binary }} create -f {{ mktemp.stdout }}/registry-claim.json
+  register: pvc_create_output
+  failed_when: ('already exists' not in pvc_create_output.stderr if pvc_create_output.stderr else False) or ('created' not in pvc_create_output.stdout if pvc_create_output.stdout else False)
+  when: attach_registry_volume | bool
```

Comment:
@abutcher with the volume mount, I don't think you have to worry about the volume already being mounted either. Kubernetes should handle the volume mounting.

Also, the pv/pvc workflow is simple in this use case. I'm not sure it'll be as simple once we add additional infra containers.  We would need to then ensure that each PV we create is claimed by the PVC associated with the right container.